runtime/nvqir/cutensornet/simulator_mps_register.cpp

```diff
@@ -65,6 +67,29 @@ class SimulatorMPS : public SimulatorTensorNetBase {
       m_relCutoff = relCutoff;
       cudaq::info("Setting MPS relative cutoff to {}.", m_relCutoff);
     }
+    static const std::unordered_map<std::string, cutensornetTensorSVDAlgo_t>
+        g_stringToAlgoEnum{{"GESVD", CUTENSORNET_TENSOR_SVD_ALGO_GESVD},
+                           {"GESVDJ", CUTENSORNET_TENSOR_SVD_ALGO_GESVDJ},
+                           {"GESVDP", CUTENSORNET_TENSOR_SVD_ALGO_GESVDP},
+                           {"GESVDR", CUTENSORNET_TENSOR_SVD_ALGO_GESVDR}};
```

Comment:
This could be a `constexpr` table stored with the keys sorted. One can use `std::lower_bound` to find the key without any difference in performance (other than the table doesn't need to be built at load time).

Reply:
I gave it a shot in https://github.com/NVIDIA/cuda-quantum/pull/1761/commits/2b6c85a8d2d2fc22573912900bd01ad920eea5b5. 
Could you please give it a second look :) ?